### PR TITLE
fix/better-k8s-docs-values

### PR DESCRIPTION
**Pull request by dabeeeenster**
Be explicit about where the k8s config changes go.

### CODE REVIEW

**Comment by dabeeeenster** [comment]
We need to add instructions to this PR in terms of how/where you store `values.yaml` changes? One for @khvn26 maybe?

**Review by matthewelwell — COMMENTED** [review]
I'd argue that this is not our responsibility to document. And a lot of this isn't really correct. You shouldn't modify the values.yml in our repository, you should add flagsmith to your chart.yml and then create your own values.yml file (as far as I understand it). 

One thing I've noticed that others do (e.g. https://github.com/jenkinsci/helm-charts), which I think we should also if we don't already (I cant see that we do), is link to the helm docs.

**Comment by dabeeeenster** [comment]
> I'd argue that this is not our responsibility to document. And a lot of this isn't really correct. You shouldn't modify the values.yml in our repository, you should add flagsmith to your chart.yml and then create your own values.yml file (as far as I understand it).
> 
> One thing I've noticed that others do (e.g. https://github.com/jenkinsci/helm-charts), which I think we should also if we don't already (I cant see that we do), is link to the helm docs.

Maybe we can link out to some resources on the solution to that. I think we shouldnt assume people are helm/k8s experts and if we can explain exactly how/where they modify this data that would help them

File: docs/docs/deployment/hosting/kubernetes.md [hunk 1]
```diff
@@ -16,7 +16,24 @@ helm install -n flagsmith --create-namespace flagsmith flagsmith/flagsmith
 kubectl -n flagsmith port-forward svc/flagsmith-frontend 8080:8080
 ```
 
-Then view `http://localhost:8080` in a browser. This will install using default options, in a new namespace `flagsmith`.
+Then view `http://localhost:8080` in a browser. This will install the chart using default options, in a new namespace
+`flagsmith`.
+
+If you want to make changes to the default setup you can make a local copy of the
```
Comment (khvn26) [inline docs/docs/deployment/hosting/kubernetes.md]: They _can_, but they don't _have to_. I'll very much prefer to emphasize that the default values file should be used for reference, not for input.

I wonder if we offered a [json schema](https://www.arthurkoziel.com/validate-helm-chart-values-with-json-schemas/) we could avoid this bit of docs altogether.

Comment (dabeeeenster) [reply docs/docs/deployment/hosting/kubernetes.md]: Can you provide a suggestion?

**Comment by dabeeeenster** [comment]
TBH I feel like we are really over-thinking this PR...

File: docs/docs/deployment/hosting/kubernetes.md [hunk 2]
```diff
@@ -16,7 +16,24 @@ helm install -n flagsmith --create-namespace flagsmith flagsmith/flagsmith
...
+[`charts/flagsmith/values.yaml`](https://github.com/Flagsmith/flagsmith-charts/blob/main/charts/flagsmith/values.yaml)file,
+modify it and then install with the `-f` flag:
```
Comment (khvn26) [inline docs/docs/deployment/hosting/kubernetes.md]:
```suggestion
Refer to the chart's default [`values.yaml`](https://github.com/Flagsmith/flagsmith-charts/blob/main/charts/flagsmith/values.yaml) file to learn which values are expected by the chart. You can use it as a reference for building your own values file:
```

File: docs/docs/deployment/hosting/kubernetes.md (hunk 1 above)
Comment (khvn26) [reply docs/docs/deployment/hosting/kubernetes.md]: See https://github.com/Flagsmith/flagsmith/pull/2222/files#r1222919381

**Review by khvn26 — APPROVED** [review]

**Comment by dabeeeenster** [comment]
WOHOO!

**Pull request merged**